databricks/sdk/credentials_provider.py

```diff
@@ -664,6 +665,86 @@ def inner() -> Dict[str, str]:
     return inner
 
 
+class ModelServingAuthProvider():
+    _MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+    @classmethod
+    def is_in_databricks_model_serving_environment(cls) -> bool:
+        """
+        Check if the code is running in Databricks Model Serving environment.
+        The environment variable set by Databricks when starting the serving container.
+        """
+        val = (
+            os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
+            # Checking the old env var name for backward compatibility. The env var was renamed once
+            # to fix a model loading issue, but we still need to support it for a while.
+            or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV") or "false")
+        return val.lower() == "true"
+
+    @classmethod
+    def should_fetch_model_serving_environment_oauth(cls) -> bool:
+        """
+        Check whether this is the model serving environment
+        Additionally check if the oauth token file path exists
+        """
+        return (cls.is_in_databricks_model_serving_environment()
```

Comment:
Could we test the different branches here? E.g. env var missing but file is present, and vice versa